lib/src/regist.c

```diff
@@ -640,12 +640,16 @@ static ChiakiErrorCode regist_parse_response_payload(ChiakiRegist *regist, Chiak
 			memcpy(host->name, header->value, len); \
 			host->name[len] = 0; \
 			continue; \
+		} else if(strcmp(key_str, "PS4-IP") ==0)    \
+		{ \
+			memcpy(host->name, regist->info.host, 16);  \
```

Comment:
This doesn't make much sense. You are saving the ip that already comes as input during the header parsing loop which doesn't have anything to do with it. Moreover, this ip may as well be `255.255.255.255` when you do a broadcast regist.

Reply:
How about this: I reduce this PR to the functionallity where morning and reg_key are read from config file, but the user has to pass the IP manually. Then the Ip-saving topic can be topic for another PR.

Reply:
Sounds good to me!